ui/perfherder/graphs/GraphTooltip.jsx

```diff
@@ -77,7 +77,10 @@ const GraphTooltip = ({
   let prevPushId;
   let pushUrl;
   if (prevFlotDataPointIndex !== -1) {
-    prevRevision = testDetails.data[prevFlotDataPointIndex].revision;
+    let firstTriggerIndex = testDetails.data.findIndex(
+      e => e.revision === dataPointDetails.revision
+    );
+    prevRevision = testDetails.data[firstTriggerIndex-1].revision;
```

Comment:
Just a question: there are any cases where `testDetails.data[firstTriggerIndex-1]` can be `undefined` ?
It's better to avoid any possible errors.

Reply:
Yes, it is. If you look at the very first data-point of the graph you'll see that there's no revision link. A way of fixing that is not showing at all that datapoint. I hope you don't think at modifying the objects that are passed intra-state.

Reply:
No, I was thinking of some error handling mechanism in case of `undefined`